distribution/helm/vector-agent/templates/configmap.yaml

```diff
@@ -6,11 +8,44 @@ metadata:
   labels:
     {{- include "libvector.labels" . | nindent 4 }}
 data:
+  {{- if .Values.customConfig }}
+  {{- if and .Values.customConfig $deprecatedConfig }}{{ fail "Can't use deprecated keys with customConfig" }}{{- end }}
+  vector.yaml: |
+{{ tpl (toYaml .Values.customConfig) . | indent 4 }}
+  {{- else if (not $deprecatedConfig) }}
+  vector.yaml: |
+    # Configuration for Vector.
+    # Docs: https://vector.dev/docs/
+    data_dir: "/vector-data-dir"
+    sources:
+      host_metrics:
+        type: host_metrics
+        filesystem:
+          devices:
+            excludes: ["binfmt_misc"]
+          filesystems:
+            excludes: ["binfmt_misc"]
+          mountpoints:
+            excludes: ["*/proc/sys/fs/binfmt_misc"]
+      internal_metrics:
+        type: internal_metrics
+      kubernetes_logs:
+        type: kubernetes_logs
+    sinks:
+      prometheus_sink:
+        type: prometheus_exporter
+        inputs: ["host_metrics", "internal_metrics"]
+        address: 0.0.0.0:9090
+  {{- end }}
+  {{- if $deprecatedConfig }}
```

Comment:
I guess an `else` branch would suffice here?

Or swap these, first checking `if $deprecatedConfig` and then in the else condition apply our new default config.

Reply:
I was hesitant to invert the checking as I wanted to match "today's" template as closely to the future (without deprecated options) template. An `else` seems like it'd be fine here though.